XiEditor/XiCore.swift

```diff
@@ -0,0 +1,46 @@
+// Copyright 2016 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+protocol XiCore {
+    func clientStarted(configDir: String?, clientExtrasDir: String?)
+    func setTheme(themeName: String)
+    func tracingConfig(enabled: Bool)
+}
+
+final class CoreRPC: XiCore {
```

Comment:
thinking out loud, this class _might_ not be necessary; if we added a `var connection: Connection { get } ` member to the `XiCore` protocol, then we could actually put all this logic in the methods for the protocol itself? I'm not _sure_ this is the best idea, but it might be worth considering?

Reply:
How would you approach putting the implementation in the methods of the protocol?
Protocol extensions? Or something else?

Reply:
Yea, I think protocol extensions are how this works in swift? Where you can say, "for types that implement this protocol, implement these methods?"

Reply:
Using protocol extensions we can provide some default implementation.. But we can't assign any state. Storing `var connection: Connection` would still require us to create a concrete `class`/`struct` which would store the `Connection` state.
Having a concrete `CoreRPC` which implements `XiCore` seems a little easier to digest 😄

Maybe the `CoreRPC` class name is not expressive enough? Do you think it has the functionality suitable for its name?

Reply:
Okay, sounds good.

The names are tricky. I think maybe `CoreRPC` should be `CoreConnection`, `Connection` should be something like `RPCSender`, and `CoreConnection` should be something like `StdoutRPCSender`? maaaaybe?